File: 122. Best Time to Buy and Sell Stock II/stock.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution{
public:
	int maxProfit(vector<int>& prices){
		int profit=0;

		for(int i=1;i<prices.size();i++){
			if(prices[i]>prices[i-1])
				profit+=prices[i]-prices[i-1];
		}
		return profit;

	}
};

int main(){
	int n;
	cin>>n;

	vector<int> prices(n);
	for(int i=0;i<n;i++){
		cin>>prices[i];
	}

	Solution obj;
	int ans= obj.maxProfit(prices);

	cout<<ans;
}

File: 122. Best Time to Buy and Sell Stock II/stock.kt

import java.util.Scanner

class Solution{

fun maxProfit(prices: IntArray): Int{
	var profit=0

	for(i in 1 until prices.size){
	if(prices[i]>prices[i-1]){
	profit+=prices[i]-prices[i-1]
	}
	}
	return profit
}
}

fun main(){
	 val scanner = Scanner(System.`in`)
	
	val n = readln().toInt()

	//val prices = IntArray(n)

	//for(i in 0 until n){
	//prices[i]=scanner.nextInt()
	//}

	val prices = readln().trim().split("\\s+".toRegex()).map{it.toInt()}.toIntArray()

	val obj = Solution()
	val ans= obj.maxProfit(prices)

	println(ans)
}